app/models/comment.rb

```diff
@@ -0,0 +1,11 @@
+# Service to download ftp files from the serve
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post, counter_cache: :comments_counter
+
+  def update_comments_counter
+    post.comments_counter = 0 if post.comments_counter.nil?
+    posts.comments_counter += 1
+    posts.save
+  end
```

Comment:
- [ ] We only need to update the `comments_counter` for one particular `post`. In the last two lines of this method, there is a reference to `posts` in the plural form. This may not work as expected.
  
  A more rails way would be doing something like this:
  ```ruby
  def update_comments_counter
    post.increment!(:comments_counter)
  end
  ```
  
  For the part where you are setting the initial counter to `0`, you may move this to your `post_controller` when creating a post. Let this model only take care of increasing the counter.
  
  ### OPTIONAL SUGGESTION
  
  We can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_coments_counter` method every time a new comment has been saved to the database, then make the `update_comments_counter` method private.

  ```ruby
    # app/models/comment.rb
    after_save :update_comments_counter

    private

    def update_comments_counter
      post.increment!(:comments_counter)
    end
  ```